lib/array_intersection.rb

```diff
@@ -1,4 +1,25 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  list1_hash = {}
+  intersection = []
+  
+  # create the hashmap for list1
+  list1.each do |number|
+    if list1_hash.include? number
+      list1_hash[number] += 1
+    else 
+      list1_hash[number] = 1
+    end
+  end
+  
+  # compare list2 to the hash -- if there's a match, put the number in the intersection array
+  
+  list2.each do |number|
+    if list1_hash.include? number
```

Comment:
If you reduce the count to 0 you shouldn't add it.  

```suggestion
    if list1_hash[number] > 0
```